kdeploy

```diff
@@ -0,0 +1,186 @@
+# may be executed remotely
+bootloader() {
+	# shellcheck disable=SC1091
+	. /etc/os-release
+
+	update-grub
+}
+
+# may be executed remotely
+sudo-function() {
+	(($#)) || { echo "Usage: sudo-function FUNC [ARGS...]" >&2; return 1; }
+	sudo bash -c "$(declare -f "$1");$(printf ' %q' "$@")"
+}
+
+usage() {
+	cat - <<-EOF
+	$SCRIPT_FILENAME - Deploy kernel builds
+	USAGE:
+	    $SCRIPT_FILENAME [OPTIONS]
+
+	Deploy Linux kernel build from the current directory to a local/remote system.
+	This command is expected to be used after building a kernel checkout and in the
+	directory where that build happened.
+
+	For remote deployment, both ssh and rsync are used - it's recommended to have
+	ssh key configured to avoid multiple authentication requests.
+
+	OPTIONS:
+	    -h, --help			Show this help
+	    -i, --include[=] GLOB	Include only specific module glob
+	    -m, --module-only		Only deploy modules, skip kernel/bootloader changes
+	    -r, --remote-host HOST	Host to be used for remote deploy via ssh/rsync
+	    -u, --remote-user USER	User to be used for remote deploy via ssh/rsync
+	    -p, --remote-port PORT	Port to be used for remote deploy via ssh/rsync
+
+	EXAMPLES:
+
+	Install on the same host:
+
+	    $ sudo $SCRIPT_FILENAME
+
+	Deploy kernel to a remote system:
+
+	    $ sudo $SCRIPT_FILENAME -r 192.168.0.1
+
+	Deploy kernel to a VM running on this host with port 4022 forwarded and with a
+	different user name (requirement: remote user must have sudo power):
+
+	    $ sudo $SCRIPT_FILENAME -p 4022 -r localhost -u test-user
+
+	Deploy only one module to a remote system:
+
+	    $ sudo $SCRIPT_FILENAME -r remote-system -m -i 'xe.ko*'
+
+	Deploy 2 modules to a remote system:
+
+	    $ sudo $SCRIPT_FILENAME -r remote-system -m -i 'xe.ko*' -i 'i915.ko*'
+EOF
+
+}
+
+parse_arg_val() {
+	local -n ret=$1
+	shift
+	local key val="" n=0
+
+	IFS="=" read -r key val <<< "$1"
+	if [[ -z "$val" && $# -gt 1 ]]; then
+		val="$2"
+		n=$((n+1))
+	fi
+
+	# shellcheck disable=SC2034
+	ret=("$n" "$key" "$val")
+}
+
+OPT_MODULE_INCLUDE=()
+OPT_MODULE_ONLY=no
+OPT_REMOTE_USER=$(whoami)
+OPT_REMOTE_PORT=22
+OPT_REMOTE_HOST=
+parse_args() {
+	local args
+
+	while [[ $# -gt 0 ]]; do
+		case  "$1" in
+		-h|--help)
+			usage
+			exit 0
+			;;
+		-i|--include)
+			parse_arg_val args "$1" "$2"
+			shift "${args[0]}"
+			OPT_MODULE_INCLUDE+=("${args[2]}")
+			;;
+		-m|--module-only)
+			OPT_MODULE_ONLY=yes
+			;;
+		-r|--remote-host)
+			parse_arg_val args "$1" "$2"
+			shift "${args[0]}"
+			OPT_REMOTE_HOST="${args[2]}"
+			;;
+		-u|--remote-user)
+			parse_arg_val args "$1" "$2"
+			shift "${args[0]}"
+			OPT_REMOTE_USER="${args[2]}"
+			;;
+		-p|--remote-port)
+			parse_arg_val args "$1" "$2"
+			shift "${args[0]}"
+			OPT_REMOTE_PORT="${args[2]}"
+			;;
+		*)
+			argerr "Unknown option $1"
+			;;
+		esac
+		shift
+	done
+	# TODO: check arg compat/defaults
+}
+
+parse_args "$@"
+
+KREL=$(make kernelrelease)
+
+if [[ -n "$OPT_REMOTE_HOST" && "$DESTDIR" == "/" ]]; then
```

Comment:
It was intentional, yes... because I want to be able to do:  `DEST=tmpinstall kdeploy ....` if I want to inspect what is the directory tree, without the script removing the temp directory.